packages/data-table/src/CdcDataTable.tsx

```diff
@@ -0,0 +1,187 @@
+  const { label, indexLabel, expanded } = table || {}
+
+  /* FILTERED DATA */
+  const filteredData = filterVizData(filters, data)
+
+  /* LOADING VARS */
+  const configLoading = configUrl && config === undefined
+  const dataLoading = data === undefined
+  /* INVALID VARS */
+  const invalidConfig = config === null
+  const invalidData = data === null
+
+  // processes initial config and sets state
+  const initConfig = (newConfig: Config) => {
+    const updatedConfig = { ...coveUpdateWorker(newConfig), ...defaults }
+    setConfig(updatedConfig)
+    setTable(updatedConfig.table)
+    setColumns(updatedConfig.columns)
+    setFilters(updatedConfig.filters)
+    setFilterIntro(updatedConfig.filterIntro)
+  }
+
+  const updateFilters = (newConfig: Config) => {
+    const { filters: newFilters } = newConfig
+    setFilters(newFilters)
+  }
+
+  // Observes changes to outermost container and changes viewport size in state
+  const resizeObserver = new ResizeObserver(entries => {
+    for (let entry of entries) {
+      let newViewport = getViewport(entry.contentRect.width)
+      setCurrentViewport(newViewport)
+    }
+  })
+  // Load data when component first mounts
+  const outerContainerRef = useCallback(node => {
+    if (node === null) return
+    resizeObserver.observe(node)
+  }, [])
+
+  useEffect(() => {
+    /* NO VALID INPUT */
+    if (!inputConfig && !configUrl) {
+      setConfig(null)
+      return
+    }
+
+    /* FULL CONFIG PROVIDED */
+    if (inputConfig) {
+      initConfig(inputConfig)
+      return
+    }
+
+    /* FETCH REQUIRED */
+    fetchConfig(configUrl)
+      .then(resConfig => initConfig(resConfig))
+      .catch(err => setConfig(null))
+  }, [inputConfig, configUrl])
+
+  useEffect(() => {
+    if (configLoading || invalidConfig) return
+
+    /* NO CONFIG DATA */
+    if (!inputData && !dataUrl) return
+
+    /* DATA PROVIDED */
+    if (inputData) {
+      const processedData = processData(inputData, dataDescription)
+      setData(processedData)
+      return
+    }
+
+    /* FETCH DATA */
+    fetchData(dataUrl)
+      .then(resData => {
+        const processedData = processData(resData, dataDescription)
+        setData(processedData)
+      })
+      .catch(() => setData(null))
+  }, [inputData, dataUrl, configLoading, invalidConfig, dataDescription])
+
+  /* HANDLE LOADING/ERROR STATES */
+  if (configLoading || dataLoading) return <Loading />
+  if (invalidConfig || invalidData) return <div>Something went wrong</div>
+
+  // initial static config combined with dynamic states
+  const configWithStates = {
+    ...config,
+    table,
+    columns,
+    data,
+    filters,
+    filterIntro
+  }
+
+  return (
+    <Layout.VisualizationWrapper
+      ref={outerContainerRef}
+      config={config}
+      isEditor={isEditor}
+      showEditorPanel={showEditorPanel}
+      currentViewport={currentViewport}
+    >
+      {/* EDITOR */}
+      {isEditor && (
```

Comment:
this section with EditorPanel Filters and DataTable seems really close to DataTableStandAlone could that have been used?